Composer/packages/server/src/models/asset/assetManager.ts

```diff
@@ -92,58 +100,79 @@ export class AssetManager {
     return ref;
   }
 
-  private async getRemoteTemplate(template: BotTemplate, destinationPath: string) {
-    // install package
-    if (template.package) {
-      const { stderr: initErr } = await execAsync(`dotnet new -i ${template.package.packageName}`);
-      if (initErr) {
-        throw new Error(initErr);
-      }
-      const { stderr: initErr2 } = await execAsync(`dotnet new ${template.id}`, {
-        cwd: destinationPath,
-      });
-      if (initErr2) {
-        throw new Error(initErr2);
+  public async copyRemoteProjectTemplateToV2(
+    templateId: string,
+    projectName: string,
+    ref: LocationRef,
+    user?: UserIdentity,
+    locale?: string
+  ): Promise<LocationRef> {
+    // user storage maybe diff from template storage
+    const dstStorage = StorageService.getStorageClient(ref.storageId, user);
+    const dstDir = Path.resolve(ref.path);
+    if (await dstStorage.exists(dstDir)) {
+      log('Failed copying template to %s', dstDir);
+      throw new Error('already have this folder, please give another name');
+    }
+    await mkdirs(dstDir, (err) => {
```

Comment:
Generally, async functions work in either Promise mode or callback mode, not both. [The docs](https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureDir.md) indicate that if you pass a callback, a promise is not returned. Testing with the CLI seems to confirm this:

```bash
> let fsx = require('fs-extra')
undefined
> fsx.mkdirs
[Function: mkdirs]
> fsx.mkdirs('foo')
Promise { <pending> }
> fsx.mkdirs('foo', () => console.log('done'))
undefined
```

Reply:
Ah good catch I changed it to use mkdirSync and added a try catch around it as seems to be the standard in other parts of the code base.

Reply:
`await mkdirs()`, without the callback, wrapped in a `try`-`catch` would also work!